test/src/test/java/hudson/util/FormFieldValidatorTest.java

```diff
@@ -152,4 +159,120 @@ public void negative() throws Exception {
             Publisher.all().remove(d);
         }
     }
+
+    @Issue("JENKINS-73404")
+    @Test
+    public void testValidationforComponents() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        p.getBuildersList().add(new ValidatingDescribable());
+        try (JenkinsRule.WebClient wc = j.createWebClient()) {
+            HtmlPage page = wc.getPage(p, "configure");
+            assertThat(page.asNormalizedText(), allOf(
+                    containsString("FormValidation: Password (empty)"),
+                    containsString("FormValidation: Password (populated)"),
+                    containsString("FormValidation: Textarea"),
+                    containsString("FormValidation: SecretTextarea (empty)"),
+                    containsString("FormValidation: SecretTextarea (populated)")));
+
+        }
+
+    }
+
+    public static class ValidatingDescribable extends Builder {
+
+        private Secret emptyPassword;
+        // give the secret some data so that it is hidden and not a regular field!
+        private Secret populatedPassword = Secret.fromString("secret!");
+        private String textarea;
+        private Secret emptySecretTextarea;
+        private Secret populatedSecretTextarea = Secret.fromString("sensitive!");;
+
+        @DataBoundConstructor
+        public ValidatingDescribable() {
+        }
+
+        public Secret getEmptyPassword() {
+            return emptyPassword;
+        }
+
+        @DataBoundSetter
```

Comment:
setters where not needed for the test, but helped with developer testing so I left them in for the next person that has to debug something in the area